humility-idol/src/lib.rs

```diff
@@ -398,14 +428,15 @@ fn call_arg(
             }
             match value {
                 IdolArgument::String(value) => {
-                    if value.len() != count {
+                    let bytes = bytes_from_str(value)?;
```

Comment:
This change will break the use case shown in https://github.com/oxidecomputer/humility/pull/342.

I'd suggest having `bytes_to_str` fall back to the previous behavior if the value isn't bracketed `[]`, i.e. directly interpreting the string as bytes if the length matches.

Reply:
I had convinced myself that nothing in hubris relied on this behavior but was obviously wrong! Thanks for catching this. I've updated the PR to allow these two representations to coexist.